tidy3d/plugins/mode/mode_solver.py

```diff
@@ -317,27 +318,410 @@ def data_raw(self) -> ModeSolverData:
+                },
+                "z",
+                "x",
+                "y",
+            ),
+        }
+
+        sel_coords_func, axial_comp, comp1, comp2 = coord_map[id_bent]
+
+        # Transform fields from Cartesian to cylindrical coordinates
+        for i, _rho in enumerate(rho):
+            sel_coords = {key: func(_rho) for key, func in sel_coords_func.items()}
+
+            # Find nearest values for the original data and populate cylindrical fields
+            fields_at_point = {
+                comp: getattr(mode_solver_data, comp).sel(sel_coords, method="nearest")
+                for comp in field_components
+            }
+
+            for field_type in ["E", "H"]:
+                fields_cylindrical[field_type + "r"][i] = (
+                    fields_at_point[field_type + comp1] * cos_theta
+                    + fields_at_point[field_type + comp2] * sin_theta
+                )
+                fields_cylindrical[field_type + "theta"][i] = (
+                    -fields_at_point[field_type + comp1] * sin_theta
+                    + fields_at_point[field_type + comp2] * cos_theta
+                )
+                fields_cylindrical[field_type + "axial"][i] = fields_at_point[
+                    field_type + axial_comp
+                ]
+
+        # Convert to xarray DataArray
+        cyn_field = {
+            name: xr.DataArray(fields_cylindrical[name], dims=dims, coords=coords)
+            for name in fields_cylindrical
+        }
+
+        return cyn_field
+
+    def rotate_structure(self) -> ModeSolverData:
+        "rotate the structures intersecting with modal plane by angle theta if bend_correction"
+        "is enabeled for bend simulations."
+
+        # Get the axis information
+        id_inj, id_mnt_plane = self.pop_axis((0, 1, 2), axis=self.normal_axis)
+        id_bend = id_mnt_plane[self.mode_spec.bend_axis]
+        _, (id_bt_u, id_bt_v) = self.pop_axis((0, 1, 2), axis=id_bend)
+
+        mnt_center = self.plane.center
+        angle_theta = self.mode_spec.angle_theta
+        angle_phi = self.mode_spec.angle_phi
+
+        # Calculate the rotation angle based on id_inj and id_bend
+        theta_map = {
+            (0, 2): -angle_theta * np.cos(angle_phi),
+            (0, 1): angle_theta * np.sin(angle_phi),
+            (1, 2): angle_theta * np.cos(angle_phi),
+            (1, 0): -angle_theta * np.sin(angle_phi),
+            (2, 1): -angle_theta * np.cos(angle_phi),
+            (2, 0): angle_theta * np.sin(angle_phi),
+        }
+        theta = theta_map.get((id_inj, id_bend), 0)
+
+        # Get the translation values based on id_bent
+        translate_coords = [0, 0, 0]
+        translate_coords[id_bt_u] = mnt_center[id_bt_u]
+        translate_coords[id_bt_v] = mnt_center[id_bt_v]
+        translate_coords = tuple(translate_coords)
+
+        rotated_structures = []
+        for structure in Scene.intersecting_structures(self.plane, self.structures):
```

Comment:
Do you mean to use the reduced_simulation object instead of the original simulation to get structures? I tried but it’s challenging to examine if the structures are correctly rotated with the reduced_simulation object.

Reply:
Hm. Ok. I think this is fine. We do warn users that using this approach means that things need to be well defined in a region around the mode plane, not just exactly at it.